### PR TITLE
[ISSUE #4257] Verify that the value of the permission is valid when updating the topic/broker config

### PR DESCRIPTION
**Pull request by ferrirW**
## What is the purpose of the change

[4257](https://github.com/apache/rocketmq/issues/4257)

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Review by caigy — APPROVED** [review]
LGTM

**Review by tsunghanjacktsai — APPROVED** [review]

**Review by tianliuliu — APPROVED** [review]
LGTM

File: client/src/main/java/org/apache/rocketmq/client/Validators.java [hunk 1]
```diff
@@ -107,4 +110,19 @@ public static void isNotAllowedSendTopic(String topic) throws MQClientException
         }
     }
 
+    public static void checkTopicConfig(final TopicConfig topicConfig) throws MQClientException {
+        if (!PermName.isValid(topicConfig.getPerm())) {
+            throw new MQClientException(ResponseCode.NO_PERMISSION,
+                String.format("brokerPermission value: %s is invalid.", topicConfig.getPerm()));
```
Comment (Oliverwqcwrw) [inline client/src/main/java/org/apache/rocketmq/client/Validators.java:116]: `brokerPermission value` should be `topicPermisson value`, and  is more accurate

**Review by Oliverwqcwrw — CHANGES_REQUESTED** [review]

File: client/src/test/java/org/apache/rocketmq/client/ValidatorsTest.java [hunk 2]
```diff
@@ -106,4 +110,63 @@ public void testIsNotAllowedSendTopic() {
...
+
+        topicConfig.setPerm(PermName.PERM_READ);
+        Validators.checkTopicConfig(topicConfig);
+
+        try {
+            topicConfig.setPerm(PermName.PERM_PRIORITY);
+            Validators.checkTopicConfig(topicConfig);
+        } catch (MQClientException e) {
+            assertThat(e.getResponseCode()).isEqualTo(ResponseCode.NO_PERMISSION);
+            assertThat(e.getErrorMessage()).isEqualTo(String.format("brokerPermission value: %s is invalid.", topicConfig.getPerm()));
```
Comment (Oliverwqcwrw) [inline client/src/test/java/org/apache/rocketmq/client/ValidatorsTest.java:131]: `brokerPermission value` should be `topicPermisson value`, and is more accurate

File: client/src/test/java/org/apache/rocketmq/client/ValidatorsTest.java [hunk 3]
```diff
@@ -106,4 +110,63 @@ public void testIsNotAllowedSendTopic() {
...
+        }
+
+        try {
+            topicConfig.setPerm(PermName.PERM_PRIORITY | PermName.PERM_WRITE);
+            Validators.checkTopicConfig(topicConfig);
+        } catch (MQClientException e) {
+            assertThat(e.getResponseCode()).isEqualTo(ResponseCode.NO_PERMISSION);
+            assertThat(e.getErrorMessage()).isEqualTo(String.format("brokerPermission value: %s is invalid.", topicConfig.getPerm()));
```
Comment (Oliverwqcwrw) [inline client/src/test/java/org/apache/rocketmq/client/ValidatorsTest.java:139]: `brokerPermission value` should be `topicPermisson value`, and is more accurate

**Review by Oliverwqcwrw — CHANGES_REQUESTED** [review]

**Comment by ferrirW** [comment]
@duhenglucky this pr can already be merged, 3q.

**Review by duhenglucky — APPROVED** [review]

**Pull request merged**